c7n/resources/route53.py

```diff
@@ -407,3 +407,47 @@ def process(self, resources, event=None):
             elif not logging and not state:
                 results.append(r)
         return results
+
+
+@HostedZone.filter_registry.register('dangling-records')
+class DanglingRecords(Filter):
+    """Filter for hosted zones that have dangling records
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: dangling-records
+            resource: hostedzone
+            filters:
+              - type: dangling-records
+    """
+
+    permissions = ('route53:ListResourceRecordSets',
+                   'route53:GetHostedZone',
+                   'ec2:DescribeAddresses')
+    schema = type_schema('dangling-records')
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('route53')
+        ec2 = local_session(self.manager.session_factory).client('ec2')
+        results = []
+
+        for r in resources:
+            zid = r['Id'].split('/', 2)[-1]
+            record_sets = self.manager.retry(
+                client.list_resource_record_sets, HostedZoneId=zid)['ResourceRecordSets']
+            record_sets = [record_set for record_set in record_sets if record_set['Type'] == 'A']
+            eips = [eip for record_set in record_sets for eip in record_set['ResourceRecords']]
```

Comment:
these aren't just eips, its common to use r53 for private records as well, i do this for my own stuff with tailscale, at min would want to ensure we're not looking at rfc1918 ips here.